### PR TITLE
Bug Fix

### PR DESCRIPTION
**Pull request by rainmaker490**
reload rows calls cell for row at index path delegate method (which re-renders the cell according to the Messages model.)
update the model to "change" the sender type. this will propagate through to the view.

### CODE REVIEW

**Comment by rainmaker490** [comment]
other nits:
1. use (nonatomic, strong)
2. update the public API to only take in models (move IBOutlets to the implementation file)
3. nits: rename setup methods for setting up UI
4. deleted some unused + commented out code

Other things not addressed 
1. naming conventions: try to spell things out (instead of saying MessagesVC, update to MessagesViewController)

File: ProjectVoyage/Cell/ChatCell.h [hunk 1]
```diff
@@ -12,12 +12,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface ChatCell : UITableViewCell
 
-@property (strong, nonatomic) IBOutlet UILabel *recipientLabel;
-@property (strong, nonatomic) IBOutlet UILabel *descriptionLabel;
-@property (strong, nonatomic) IBOutlet UIImageView *recipientImage;
-@property (weak, nonatomic) IBOutlet UILabel *dateChat;
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/ChatCell.h:18]: we want to ensure that all IBOutlets (unless they create a retain cycle) are strong

Comment (rainmaker490) [reply ProjectVoyage/Cell/ChatCell.h:18]: more about retain cycles: https://www.cocoawithlove.com/2009/07/rules-to-avoid-retain-cycles.html

File: ProjectVoyage/Cell/ChatCell.m [hunk 2]
```diff
@@ -8,27 +8,21 @@
...
-- (void)awakeFromNib {
-    [super awakeFromNib];
-    // Initialization code
-}
-
-
-- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
-    [super setSelected:selected animated:animated];
-    // Configure the view for the selected state
-}
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/ChatCell.m:22]: We don't want to commit code unless there is logic here that needs to be overriden

File: ProjectVoyage/Cell/ChatCell.m [hunk 3]
```diff
@@ -8,27 +8,21 @@
...
 
+@implementation ChatCell
 
 - (void) setChat:(Chat *)chat {
     _chat = chat;
-    self.recipientImage.image = [UIImage imageWithData:[chat.recipientImage getData]];
-    self.recipientLabel.text = chat.recipientName;
-    self.descriptionLabel.text = chat.chatDescription;
-    
-    self.dateChat.text = [Util formatDateString:chat.createdAt];
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/ChatCell.m:31]: just a nit

in objective c, you can use the underscore infront of ivars to reference them

File: ProjectVoyage/Cell/MessageCell.h [hunk 4]
```diff
@@ -19,18 +19,7 @@ for incoming or outgoing messages
...
-
-@property (weak, nonatomic) id <ContainerProtocol> delegate;
-
-// Container
-@property (strong, nonatomic) UILabel *messageType;
-@property (strong, nonatomic) UIButton *deleteButton;
-@property (strong, nonatomic) UIButton *editButton;
-@property (strong, nonatomic) UIButton *changeSenderButton;
-@property (strong, nonatomic) UITextView *textView;
-@property (strong, nonatomic) UIImageView *bubbleImage;
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/MessageCell.h:31]: These should be moved to the implementation file (the public API should be as clean as possible)

File: ProjectVoyage/Cell/MessageCell.m [hunk 5]
```diff
@@ -10,10 +10,13 @@
 #import "Message.h"
 
 @interface MessageCell ()
-// Message
-// @property (strong, nonatomic) UILabel *timeLabel;
-
-
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/MessageCell.m:16]: unused code

File: ProjectVoyage/Cell/MessageCell.m [hunk 6]
```diff
@@ -82,23 +90,22 @@ - (void) setMessage:(Message *)message {
 }
 
 - (void) buildCell {
-    [self setTextView];
-    [self setBubble];
-    [self setMessageType];
+    [self setupTextView];
+    [self setupBubbleView];
+    [self setupMessage];
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/MessageCell.m:95]: this should probably be setupMessageView

File: ProjectVoyage/Cell/MessageCell.m [hunk 7]
```diff
@@ -163,19 +170,31 @@ - (void) setTextView {
     textView_y = -3;
     autoresizing = UIViewAutoresizingFlexibleLeftMargin;
     
-    /* if sender is someone else {
-        textView_x = 20;
-        textView_y = -1;
-        autoresizing = UIViewAutoresizingFlexibleRightMargin;
-    */
-
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/MessageCell.m:171]: unused code :)

File: ProjectVoyage/Cell/MessageCell.m [hunk 8]
```diff
@@ -163,19 +170,31 @@ - (void) setTextView {
...
+        CGFloat bubble_y = 0;
+        CGFloat bubble_width;
+        CGFloat bubble_height = _textView.frame.size.height + 8;
+        
+        bubble_x = marginRight;
+        bubble_width = _textView.frame.origin.x + _textView.frame.size.width + marginLeft;
+        
+        _bubbleImage.frame = CGRectMake(bubble_x, bubble_y, bubble_width, bubble_height);
+        _bubbleImage.autoresizingMask = _textView.autoresizingMask;
+    }
```
Comment (rainmaker490) [inline ProjectVoyage/Cell/MessageCell.m:192]: I haven't vetted the code inside this if statement, (just copied from the view controller).

use the message cell to render your views based on the Message object. this is all this class should be responsible for. The view controller will be responsible for the logic

**Review by gdavidss — APPROVED** [review]

**Pull request merged**